tutorials/auto-data-tokenize/index.md

```diff
@@ -0,0 +1,446 @@
+--stagingLocation="gs://${TEMP_GCS_BUCKET}/staging" \
+--workerMachineType="n1-standard-1" \
+--sampleSize=500 \
+--fileType="AVRO" \
+--inputFilePattern="gs://${TEMP_GCS_BUCKET}/userdata.avro" \
+--reportLocation="gs://${TEMP_GCS_BUCKET}/dlp_report/"
+```
+
+The pipeline detects for all the standard information types supported by DLP.
+Use `--observableInfoTypes` to provide custom info-types you need to detect.
+
+### Pipeline DAG
+
+![Sampler Pipeline DAG](sampler_pipeline_dag.png)
+
+### Retrieve report
+
+The sampling pipeline outputs the Avro schema (or converted for Parquet) of the files and columnn report with sensitive
+information types.
+
+```shell script
+mkdir -p dlp_report/ && rm dlp_report/*.json
+gsutil -m cp "gs://${TEMP_GCS_BUCKET}/dlp_report/*.json" dlp_report/
+```
+
+List all the column names that have been identified.
+
+```shell script
+cat dlp_report/col-*.json | jq .columnName
+```
+The output will match the following list.
+
+```text
+"$.kylosample.birthdate"
+"$.kylosample.cc"
+"$.kylosample.email"
+"$.kylosample.first_name"
+"$.kylosample.ip_address"
+"$.kylosample.last_name"
+```
+
+You can view the details of the identified column by issuing `cat` command for the file.
+
+```shell script
+cat dlp_report/col-kylosample-cc-00000-of-00001.json
+```
+
+Following is a snippet of one such column.
+```json
+{
+  "columnName": "$.kylosample.cc",
+  "infoTypes": [
+    {
+      "infoType": "CREDIT_CARD_NUMBER",
+      "count": "394"
+    }
+  ]
+}
+```
+
+> The `"count"` value will vary based on the randomly selected samples.
+
+## Launch tokenizing pipeline
+
+The sampling pipeline used a small number of samples from the original dataset to identify sensitive informaiton using
+DLP. The tokenizing pipeline would process the entire dataset and encrypt the desired columns using the provided Data
+Encryption Key (DEK).
+
+```shell script
+tokenize_pipeline --project="${PROJECT_ID}" \
+--region="${REGION_ID}" \
+--runner="DataflowRunner" \
+--serviceAccount=${DLP_RUNNER_SERVICE_ACCOUNT_EMAIL} \
+--workerMachineType="n1-standard-1" \
+--schema="$(<dlp_report/schema.json)" \
+--tinkEncryptionKeySetJson="$(<${WRAPPED_KEY_FILE})" \
+--mainKmsKeyUri="${MAIN_KMS_KEY_URI}" \
+--fileType="AVRO" \
+--inputFilePattern="gs://${TEMP_GCS_BUCKET}/userdata.avro" \
+--outputDirectory="gs://${TEMP_GCS_BUCKET}/encrypted/" \
+--tokenizeColumns="$.kylosample.cc" \
+--tokenizeColumns="$.kylosample.email"
+```
+
+The pipeline executes asynchronously on Dataflow. You can check the progress by following the JobLink printed
+```
+INFO: JobLink: https://console.cloud.google.com/dataflow/jobs/<your-dataflow-jobid>?project=<your-project-id>
+```
+![Encrypting Pipeline DAG](encryption_pipeline_dag.png)
+
+### Verify encrypted result
+
+Load the output files into BigQuery to verify that `tokenizeColumns` have been encrypted.
+
+1. Create a BigQuery dataset for tokenized data
+   Replace <i><bigquery-region></i> with a region of your choice.
```

Comment:
should we highlight that the dataset must be in the same regional or multi-regional location as the Cloud Storage bucket or else the bq command fails?

Reply:
ACK. Looking for a link to the issue.